stdlib/src/mmap.rs

```diff
@@ -340,7 +338,8 @@ mod mmap {
                 }
             };
 
-            if fd != -1 {
+            let fd = unsafe { crt_fd::Borrowed::try_borrow_raw(fd) }.ok();
+            if let Some(fd) = fd {
```

Comment:
```suggestion
            let fd = unsafe { crt_fd::Borrowed::try_borrow_raw(fd) };
            if let Ok(fd) = fd {
```
No benefit for conversion